packages/components/src/progress-bar/index.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import type { ProgressBarProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+// Width of the indicator for the indeterminate progress bar
+export const INDETERMINATE_TRACK_WIDTH = 50;
+
+export function ProgressBar(
+	props: WordPressComponentProps< ProgressBarProps, 'div', false >
+) {
+	const { className, value, indicatorColor, trackColor } = props;
+	const isIndeterminate = ! Number.isFinite( value );
+	const wrapperClasses = classnames( 'components-progress-bar', className, {
+		'is-indeterminate': isIndeterminate,
+	} );
+	const trackStyle = {
+		backgroundColor: trackColor ? trackColor : undefined,
+	};
+	const indicatorStyle = {
+		width: `${ isIndeterminate ? INDETERMINATE_TRACK_WIDTH : value }%`,
+		backgroundColor: indicatorColor ? indicatorColor : undefined,
+	};
+
+	return (
+		<div className={ wrapperClasses } style={ trackStyle }>
+			<div
+				className="components-progress-bar__indicator"
+				style={ indicatorStyle }
+			/>
+			<progress max={ 100 } value={ value } />
```

Comment:
I guess you don't use the `progress` element because of the non-standard colors, right?

If we hide it anyway, why would we need render it? 🤔

Reply:
> I guess you don't use the progress element because of the non-standard colors, right?

Yep, that's why 😉 Plus it's difficult to style it well for all browsers and that has its own limitations in terms of what and how we can animate. While with custom elements we can do whatever we want in terms of functionality and styling, while still maintaining 100% browser compatibility. 

> If we hide it anyway, why would we need render it? 🤔

For accessibility reasons. Which is why I changed it to be rendered, but invisible in https://github.com/WordPress/gutenberg/pull/53030/commits/f1d5c809214c6df3e36c0c49aee6f7c18df0ba28. The idea is that we don't need to manually specify all the roles and aria attributes, they're handled natively by the browser because we're using the native element.

I'm not sure how much this makes sense from an accessibility PoV though. cc @andrewhayward for feedback - how can we make sure the progress bar component is accessible?

Reply:
Ideally the `<progress>` element should have a label of some sort (probably `aria-label`) as well as either a `ref` or `id` so that containers can point to it...

```
<Container aria-busy={true} aria-describedby={progressBarId} />
<ProgressBar id={progressBarId} label="Loading..." />
```

Reply:
Thanks for your feedback @andrewhayward!

I've added an `aria-label` in d6d70445073a250ed166b1d44892189415d8ac39.

Regarding the `ref/id`, there are different means to get there - we could use refs with `forwardRef`, or we could use an additional `id` prop that's passed as an `id` attribute of the `progress` element. 

Can you demonstrate an example existing usage so I know which one to go with?

Reply:
A lot of controls actually employ both methods, [`TextControl` for example](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text-control/index.tsx).

```js
function UnforwardedControl ({ props, ref }) {
  const { ..., id: idProp, ... } = props;
  const id = useInstanceId( Control, 'control', idProp );

  return (
    <div>
      ...
      <el ref={ref} id={id} .../>
      ...
    </div>
  );
}

export const Control = forwardRef( UnforwardedControl );
```

This allows consumers to operate in either way. They can provide their own ID, based on their own use-case, or they can use a ref and take the generated ID instead.

Reply:
Thanks for confirming, sounds like a plan 👍 

Implemented in f772032e78d0f3b80bcb5b35b1959415d58bc0e4